### PR TITLE
refactor(pricing): support pricing for multi-leg trips

### PR DESCRIPTION
**Pull request by GoodluckH**

### CODE REVIEW

File: server/src/engine.rs [hunk 1]
```diff
@@ -1,33 +1,102 @@
...
 }
 
+/// Check that all pricing requests have the same service type.
+fn check_pricing_requests(requests: &[PricingRequest]) -> Result<(), PricingError> {
+    if requests.is_empty() {
+        return Err(PricingError::NoRequests);
+    }
+    let service_type = requests[0].service_type;
+    for request in requests {
+        if request.service_type != service_type {
```
Comment (ServiceDog) [inline server/src/engine.rs:89]: This is fine for now -

Do you think we'll ever have flights that are carrying cargo _and_ passengers? If there's passenger seating and a small cargo hold, for instance, for people and objects going to the same vertiport

File: server/src/main.rs [hunk 2]
```diff
@@ -28,17 +28,22 @@ impl Pricing for ArrowPricing {
...
-        let response = pricing_grpc::PricingResponse { price };
-        grpc_debug!("Returning response: {:?}", response);
-        grpc_info!("Successfully computed pricing; returning response");
-        Ok(tonic::Response::new(response))
+        info!("Received pricing request");
+        let prices = get_pricing(query);
+        match prices {
+            Ok(prices) => {
+                let total_price = prices.iter().fold(0.0, |acc, x| acc + x);
+                let response = pricing_grpc::PricingResponse { price: total_price };
```
Comment (ServiceDog) [inline server/src/main.rs:39]: Might we want to display per-leg cost breakdown to the customer? Is that too much?

**Review by ServiceDog — DISMISSED** [review]

File: server/src/engine.rs (hunk 1 above)
Comment (ServiceDog) [reply server/src/engine.rs:89]: I suppose this calculation is for a specific customer order, and their order shouldn't turn from a cargo mission to a passenger mission between legs. Ignore my comment :)

File: server/src/main.rs (hunk 2 above)
Comment (GoodluckH) [reply server/src/main.rs]: [18a8e13](https://github.com/Arrow-air/svc-pricing/pull/17/commits/18a8e139860ee66001dda4ad87e896e02fff3eac)

File: server/src/engine.rs (hunk 1 above)
Comment (GoodluckH) [reply server/src/engine.rs:89]: yea this is purely to prevent any programming errors when making rpc calls to pricing.

**Review by ServiceDog — APPROVED** [review]

**Pull request merged**